hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/protocolPB/OzoneManagerProtocolServerSideTranslatorPB.java

```diff
@@ -158,7 +158,7 @@ private OMResponse processRequest(OMRequest request) throws
         // if current OM is leader and then proceed with processing the request.
         if (request.hasS3Authentication()) {
           s3Auth = true;
-          checkLeaderStatus();
+          OzoneManagerRatisUtils.checkLeaderStatus(ozoneManager);
           S3SecurityUtil.validateS3Credential(request, ozoneManager);
         }
```

Comment:
With `validateS3Credential()` checking the leader status when validating the password (S3Token) from the `OMRequest`, is the `OzoneManagerRatisUtils.checkStatus()` in the line above it unnecessary, redundant?

Reply:
Correct me if I'm wrong. With HDDS-4440 and HDDS-5881, S3 authentication will be carried as part of the OMRequest.  It doesn't leverage the underline hadoop RPC SASL authentication,  and these requests will not be processed by SASL.  So the leader check is still needed here.